### PR TITLE
docker-compose for testing

### PR DESCRIPTION
**Pull request by mfrdrcks**
Added a docker-compose for ease of testing.
Added a section to the README about using the docker-compose.
Added a config.json that is set for easy use of docker-compose in a non-secure manner.

This relies on the ANONYMOUS_SUBMISSION that has been added in the add-registration branch.

To test:
Clone repo
Run `docker-compose up --build`
Attempt to access a new local catalog at localhost:5000 in your web browser
The catalog home page should be there, though with no data
Submit an extractor.
Stop the docker-compose.
Restart the docker-compose using `docker-compose up`
Attempt to access the local catalog (with the previously added extractor) at localhost:5000
Note that the volume seems to empty every time `docker-compose up --build` is called
This can be avoided by building the transformation-catalog image in a separate command, and simply calling `docker-compose up` after making changes to the image

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    image: mongo:4.2
+    volumes:
+      - "dbdata:/data/db"
+  catalog:
+    ports:
+    - "5000:5000"
+    volumes:
+    - ./instance:/app/instance
+    build: .
+    image: transformations-catalog
```
Comment (robkooper) [inline docker-compose.yml]: this should be maybe clowder/catalog, or clowder/transformations-catalog (I like the first one). At some point we should push this image.

File: instance/config.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "TRANSFORMATIONS_DATABASE_URI": "mongodb://mongodb:27017",
+  "TRANSFORMATIONS_DATABASE_NAME": "transformations_catalog_test",
```
Comment (robkooper) [inline instance/config.json:3]: Maybe shorten this to catalog or transformations_catalog (same as image name). No need for _test.

**Review by robkooper — CHANGES_REQUESTED** [review]

File: instance/config.json (hunk 2 above)
Comment (mfrdrcks) [reply instance/config.json:3]: I chose this specifically to not clash with any other databases that might be in use. The name is long, but an easy change (catalog_test?). Though as a sample stripped down config file, maybe using the suggested name makes sense.

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,3 @@ __pycache__
 venv/
 .DS_Store
 instance/config.py
-instance/config.json
```
Comment (lmarini) [inline .gitignore]: Not sure why this was removed.

**Review by lmarini — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (mfrdrcks) [reply .gitignore]: Looking back, I'm not sure either. There isn't anything particularly sensitive there, but no need to let someone accidentally upload their own setup. I'll get that readded.

Comment (mfrdrcks) [reply .gitignore]: I found the reason that I removed this from the .gitignore. By including this with the repo we make the docker-compose up command work out of the box with the mongodb docker started by the compose file. Removing this from the repo means that users then have to go in and modify the config.json to use the docker-compose version of mongo (if they want a completely stand alone test environment). So it was added as an easy way to get everything up and running.

**Review by lmarini — APPROVED** [review]

**Review by robkooper — APPROVED** [review]

File: instance/config.json (hunk 2 above)
Comment (robkooper) [reply instance/config.json:3]: problem is that people might just use this as the default when deploying it in production as well.

Comment (mfrdrcks) [reply instance/config.json:3]: True. I wanted to include a default config.json so that testing is as easy as possible, but that does mean that it will be used as a default for everyone. I like having the test appended for identification of changing the defaults, but it's not necessary.
